lib/user_comm_screener.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+# There are various ways within Discourse that a user can prevent
+# other users communicating with them. The purpose of this class is to
+# find which of the target users are ignoring, muting, or preventing
+# private messages from the acting user, so we can take alternative
+# action (such as raising an error or showing a helpful message) if so.
+#
+# Users may Mute another user (the actor), which will:
+#
+#   * Prevent PMs from the actor
+#   * Prevent notifications from the actor
+#
+# Users may Ignore another user (the actor), which will:
+#
+#   * Do everything that Mute does as well as suppressing content made by
+#     the actor (such as posts) from the UI
+#
+# Users may also either:
+#
+#   a) disallow all PMs sent to them or
```

Comment:
```suggestion
#   a) disallow PMs from being sent to them or
```
Minor correction since the setting prevents the PMs from being created in the first place.